ebos/equil/initstateequil.hh

```diff
@@ -1721,10 +1734,42 @@ private:
         this->temperature_ = eclState.fieldProps().get_double("TEMPI");
     }
 
+    template <class RMap>
+    void updateInitialSaltConcentration_(const Opm::EclipseState& eclState, const RMap& reg, const Grid& grid)
+    {
+        const int numEquilReg = rsFunc_.size();
+        saltVdTable_.resize(numEquilReg);
+        const auto& tables = eclState.getTableManager();
+        const Opm::TableContainer& saltvdTables = tables.getSaltvdTables();
+
+        // If no saltvd table is given, we create a trivial table for the density calculations
+        if (saltvdTables.empty()) {
+            std::vector<double> x = {0.0,1.0};
+            std::vector<double> y = {0.0,0.0};
+            for (size_t i = 0; i < numEquilReg; ++i) {
+                saltVdTable_[i].setXYContainers(x,y);
+            }
```

Comment:
You don't actually need the index here&mdash;and as written it produces a "signed-vs-unsigned" comparison [warning](https://ci.opm-project.org/job/opm-simulators-PR-builder/1130/gcc/new/fileName.1081815698/) because `numEquilReg` is `int`&mdash;since you could equally well write this loop as
```C++
for (auto& table : this->saltVdTable_) {
    table.setXYContainers(x, y);
}
```